pkg/scheduler/scheduler_test.go

```diff
@@ -1651,6 +1681,563 @@ func TestSchedule(t *testing.T) {
 				"eng-gamma/gamma4":   *utiltesting.MakeAdmission("eng-gamma").Assignment(corev1.ResourceCPU, "on-demand", "20").Obj(),
 			},
 		},
+		"multiple preemptions without borrowing": {
+			// While requiring the same shared FlavorResource (Default, cpu),
+			// multiple workloads are able to issue preemptions on workloads within
+			// their own CQs in a single scheduling cycle.
+			//
+			// Note: this test case passes for legacy logic when
```

Comment:
Interesting... how does it overcome this check? `(mode == flavorassigner.Preempt && cycleCohortsSkipPreemption.Has(cq.Cohort.Name))`?

Reply:
this line - when usage is equal to nominal, we have 0 usage. We are checking that usage is > 0, rather than a membership check - a subtle change in https://github.com/kubernetes-sigs/kueue/pull/2622/files#r1684180659 which I'm reverting in #2646.